modules/le_log/le_log.cpp

```diff
@@ -0,0 +1,93 @@
+#include "le_log.h"
+#include "le_core/le_core.h"
+
+#include <atomic>
+#include <mutex>
+#include <unordered_map>
+
+struct le_log_module_o {
+	std::string     name      = "DEFAULT";
+	std::atomic_int log_level = 1;
+};
+
+static le_log_module_o *le_log_module_default() {
+	static auto module = new le_log_module_o();
```

Comment:
I now realize that this won't work with hot reloading

Reply:
It would be great to have `le_log` as a separate pull request - it will make it easier to test things, and to make sure everything works when hot-reloading. 

Note that we want c-style interfaces for our modules, and we're not allowed to include header files (apart from `le_core.h` and `stdint.h`) in header files. 

This means that any c++ facades which may live in the header files must be pretty lightweight too, and shouldn't use any advanced c++ features which depend on including headers... 

__Some c++ish things still work without fancy c++ headers:__

* since `std::forward<T>` is nothing but syntactic sugar for an unconditional cast to an r-value, we can substitute with: `static_cast<T&&>`
* `std::string` can in nearly all cases be substituted by `char const *`, where needed, we may also pass a `size_t` to contain the length of a string.

Reply:
ok, let's focus on le_log first. I'll try the non sugary way and open a new PR. Still wrapping my head around on how to solve something like persistent static data.

Reply:
Great! Persistent static data? My suggestion would be to first see if things can be made to work without. 

Then, maybe take a look at `texture_handle_library` works in `le_renderer.cpp`, and see if something like that might help.

Reply:
Normally I'd strongly search for any other solution. But in a logging context it might make sense. Basically I want a global list of named loggers where each log level can be set individually. and to top things off they also need to be threadsafe.